modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4ReleaseChannelFutureListener.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import org.elasticsearch.common.lease.Releasable;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+
+/**
+ * A channel listener that releases a {@link org.elasticsearch.common.lease.Releasable} when
+ * the operation is complete.
+ */
+class Netty4ReleaseChannelFutureListener implements ChannelFutureListener {
```

Comment:
this looks like we could just use a lambda where this is used? - it's only one place anyway